### PR TITLE
Documentation - Updated FieldPanel('title'...) to use TitleFieldPanel('title'...) in docs folder

### LINKED ISSUE #11078: Documentation - Update all `FieldPanel('title'...` examples to use `TitleFieldPanel('title'...`
Wagtail 5.0.2 introduced a new recommended `TitleFieldPanel` that supports the linking for the slug field and other UI improvements for the title field. However, many of our documentation examples do not use this and it could cause confusion if the slug field is not working as expected.

- https://docs.wagtail.org/en/stable/releases/5.0.2.html#use-of-titlefieldpanel-for-the-page-title-field
- https://docs.wagtail.org/en/stable/reference/pages/panels.html#title-field-panel

### Pertinent section of the Wagtail docs

There are a few places, listed below, but generally where we have `FieldPanel('title'...` set, we would want this to be updated to `TitleFieldPanel('title'...`.  A reminder that not all title fields used in the documentation are for `Page` models, so we cannot just bulk update, we should also update the imports in code snippets if applicable.

#### Example `docs/advanced_topics/customisation/page_editing_interface.md`

```diff
- from wagtail.admin.panels import TabbedInterface, ObjectList
+ from wagtail.admin.panels import TabbedInterface, TitleFieldPanel, ObjectList

class BlogPage(Page):
    # field definitions omitted

    content_panels = [
-        FieldPanel('title', classname="title"),
+        TitleFieldPanel('title', classname="title"),
        FieldPanel('date'),
```

### Details

The following files would need to be updated.

- `docs/advanced_topics/customisation/page_editing_interface.md`, see `content_panels` & `shared_panels` in code examples (3 all up).
- `docs/reference/pages/panels.md` see `Placeholder text` section ~L380
- There may be more, we do not want to update any of the ModelAdmin docs as these are all non-page models, even though they have a 'title' field for some of those models.

### PR DESCRIPTION
**Pull request by Chinex-Boroja**
Fixes #11078 

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [x] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: docs/advanced_topics/customisation/page_editing_interface.md [hunk 1]
```diff
@@ -32,13 +32,13 @@ class BlogPage(Page):
...
-from wagtail.admin.panels import TabbedInterface, ObjectList
+from wagtail.admin.panels import TabbedInterface, TitleFieldPanel, ObjectList
 
 class FundingPage(Page):
     # field definitions omitted
 
     shared_panels = [
-        FieldPanel('title', classname="title"),
+        TitleFieldPanel('title', classname="title"),
         FieldPanel('date'),
```
Comment (lb-) [inline docs/advanced_topics/customisation/page_editing_interface.md:42]: There was one more further down on this page but I have added it. Thank you.

**Review by lb- — APPROVED** [review]
Thank you @Chinex-Boroja  - I have made one addition but the rest look good.

I have credited you as `Chinedu Ihedioha` on our contributors list but please let me know if you want a different name listed.

**Comment by Chinex-Boroja** [comment]
> Thank you @Chinex-Boroja - I have made one addition but the rest look good.
> 
> I have credited you as `Chinedu Ihedioha` on our contributors list but please let me know if you want a different name listed.

Thank you @lb-, the name is perfectly fine. Thanks for the addition too, I didn't catch it. If there are similar issues like this, I would also like to work on them.

**Comment by lb-** [comment]
Thanks @Chinex-Boroja  - I cannot think of anything specifically but here's an ordered list of documentation items that may be good to look at https://github.com/wagtail/wagtail/issues?q=is%3Aopen+is%3Aissue+label%3ADocumentation+sort%3Aupdated-desc

**Pull request merged**